iaso/api/tasks/__init__.py

```diff
@@ -70,3 +79,211 @@ def get_queryset(self):
         profile = self.request.user.iaso_profile
         order = self.request.query_params.get("order", "created_at").split(",")
         return Task.objects.filter(account=profile.account).order_by(*order)
+
+
+class ExternalTaskSerializer(TaskSerializer):
+    def update(self, task, validated_data):
+        has_progress_message = validated_data.get("progress_message", None) is not None
+        has_status = validated_data.get("status", None) is not None
+        has_progress_value = validated_data.get("progress_value", None) is not None
+        has_end_value = validated_data.get("end_value", None) is not None
+        if (has_status or has_progress_value or has_progress_message or has_end_value) and not task.external:
+            raise serializers.ValidationError({"external": "Cannot modify non external tasks"})
+        if validated_data.get("should_be_killed", None) is not None:
+            task.should_be_killed = validated_data["should_be_killed"]
+            if validated_data["should_be_killed"]:
+                task.status = KILLED
+        if has_status:
+            task.status = validated_data["status"]
+            if (
+                validated_data["status"] == SUCCESS
+                or validated_data["status"] == ERRORED
+                or validated_data["status"] == KILLED
+            ):
+                task.ended_at = datetime.now()
+        if has_progress_message:
+            task.progress_message = validated_data["progress_message"]
+        if has_progress_value:
+            task.progress_value = validated_data["progress_value"]
+        if has_end_value:
+            task.end_value = validated_data["end_value"]
+        task.save()
+        return task
+
+
+class ExternalTaskPostSerializer(serializers.Serializer):
+    slug = serializers.CharField()
+    config = data = serializers.JSONField()
+    id_field = data = serializers.JSONField()
+
+    def validate(self, attrs):
+        validated_data = super().validate(attrs)
+        request = self.context["request"]
+        slug = request.data.get("slug", None)
+        config = request.data.get("config", None)
+        id_field = request.data.get("id_field", None)
```

Comment:
Are these fields optional ?

Reply:
only `slug` is always mandatory.
`id_field` is used when using OpenHexa API to check for active runs for a specific country, in this case `id_field` is `{country_id:12646}`, but we may not always have to perform such a specific check

`config` is a dict regrouping the arguments of that need to be passed to the pipeline. It's possible to have pipeline that don't take arguments